### PR TITLE
Add tests in path_provider and url_launcher plugins

### PR DESCRIPTION
**Pull request by szakarias**
+ some missing copyright statements

### CODE REVIEW

File: packages/path-provider/test/path_provider_test.dart [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      log.add(methodCall);
+      return response;
+    });
+
+    Directory directory = await getTemporaryDirectory();
+
+    expect(log, equals(<MethodCall>[new MethodCall('getTemporaryDirectory')]));
+    expect(directory, isNull);
+    log.clear();
+
```
Comment (mravn-google) [inline packages/path-provider/test/path_provider_test.dart]: We start a new test here. Should ideally be separated into two invocations of `test`.

**Comment by szakarias** [comment]
Thanks. Could you please have another quick look?

File: packages/path-provider/test/path_provider_test.dart [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    expect(directory, isNull);
+  });
+
+  test('getApplicationDocumentsDirectory test', () async {
+    response = null;
+    Directory directory = await getApplicationDocumentsDirectory();
+    expect(
+        log,
+        equals(
+            <MethodCall>[new MethodCall('getApplicationDocumentsDirectory')]));
```
Comment (mravn-google) [inline packages/path-provider/test/path_provider_test.dart]: A trailing comma may make `dartfmt` do something prettier.

File: packages/url-launcher/test/url_launcher_test.dart [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  tearDown(() {
+    log.clear();
+  });
+
+  test('canLaunch test', () async {
+    await canLaunch('http://example.com/');
+    expect(
+        log,
+        equals(
+            <MethodCall>[new MethodCall('canLaunch', 'http://example.com/')]));
```
Comment (mravn-google) [inline packages/url-launcher/test/url_launcher_test.dart]: Same here.

**Pull request merged**